packages/create-cloudflare/src/__tests__/templates.test.ts

```diff
@@ -246,3 +249,32 @@ describe("addWranglerToGitIgnore", () => {
 		);
 	}
 });
+
+describe("downloadRemoteTemplate", () => {
+	function mockDegit() {
+		// @ts-expect-error only clone will be used
+		return vi.mocked(degit).mockReturnValue({
+			clone: () => Promise.resolve(),
+		});
+	}
+
+	afterEach(() => {
+		vi.resetAllMocks();
+	});
+
+	test("should download template using degit", async () => {
+		const mock = mockDegit();
+
+		await downloadRemoteTemplate("cloudflare/workers-sdk");
+
+		expect(mock).toBeCalled();
+	});
+
+	test("should not use a spinner", async () => {
```

Comment:
This needs a better test name or a comment block explaining why.

This test is highly coupled to the code so clarifying it is a regression test and how the regression occurred and how not using spinner avoids it in the future will be more helpful.

Think about this test failing in the future and the dev wondering why they should not use a spinner

Reply:
Notably, this test has made it into this PR for fixing "cloning non-existent repository URL". But it also seems this PR is fixing "cloning a private repo" which requires prompting for username/password. That should also be included in the changeset